nova/tests/unit/scheduler/filters/test_cpu_info_migration_filters.py

```diff
@@ -0,0 +1,164 @@
+#    under the License.
+import mock
+
+from oslo_serialization import jsonutils
+
+from nova.exception import ComputeHostNotFound
+from nova import objects
+from nova.objects.compute_node import ComputeNode
+from nova.scheduler.filters import cpu_info_migration_filter
+from nova import test
+from nova.tests.unit.scheduler import fakes
+
+
+class TestCpuFlagsMigrationFilter(test.NoDBTestCase):
+    SOURCE_NODE = fakes.COMPUTE_NODES[0]
+
+    def setUp(self):
+        super(TestCpuFlagsMigrationFilter, self).setUp()
+        self.filter = cpu_info_migration_filter.CpuInfoMigrationFilter()
+        self._default_scheduler_hints = dict(
+            source_host=[self.SOURCE_NODE.host],
+            source_node=[self.SOURCE_NODE.hypervisor_hostname],
+        )
+
+    @mock.patch.object(cpu_info_migration_filter, 'get_admin_context')
+    @mock.patch.object(cpu_info_migration_filter, 'request_is_resize')
+    @mock.patch.object(cpu_info_migration_filter, 'request_is_rebuild')
+    @mock.patch.object(ComputeNode, 'get_by_host_and_nodename')
+    def _test_filter_all(self, mock_get_cn, mock_is_rebuild, mock_is_resize,
+            mock_get_admin_context, compute_node=None, is_resize=False,
+            is_rebuild=False, hosts=None, filtered_hosts=None,
+            **scheduler_hints):
+        if not scheduler_hints:
+            scheduler_hints = self._default_scheduler_hints
+        hosts = mock.sentinel.hosts if hosts is None else hosts
+        if filtered_hosts is None:
+            filtered_hosts = hosts
+        admin_ctx = mock.sentinel.admin_ctxt
+        mock_get_admin_context.return_value = admin_ctx
+        mock_is_rebuild.return_value = is_rebuild
+        mock_is_resize.return_value = is_resize
+        mock_get_cn.side_effect = compute_node
+        spec_obj = objects.RequestSpec(
+                        context=mock.sentinel.ctx,
+                        scheduler_hints=scheduler_hints)
+
+        result = self.filter.filter_all(hosts, spec_obj)
+
+        try:
+            # In case the filter returns an iterable
+            result = list(result)
+        except TypeError:
+            pass
+
+        self.assertEqual(result, filtered_hosts)
+        if is_resize:
+            mock_is_resize.assert_called_with(spec_obj)
+        if is_rebuild:
+            mock_is_rebuild.assert_called_with(spec_obj)
+        if compute_node is None:
+            mock_get_admin_context.assert_not_called()
+            mock_get_cn.assert_not_called()
+        else:
+            mock_get_admin_context.assert_called()
+            source_host = spec_obj.get_scheduler_hint('source_host')
+            source_node = spec_obj.get_scheduler_hint('source_node')
+            mock_get_cn.assert_called_with(admin_ctx, source_host, source_node)
+        return spec_obj
+
+    def _test_filter_all_no_host_passes(self, **kwargs):
+        with mock.patch.object(
+                self.filter, '_is_superset') as mock_host_passes:
```

Comment:
Update to `_are_cpu_flags_supported`

Reply:
Sorry, I did run the unit tests but only for the vmware part by routine. Fixed it